scripts/Testing/Testcases/HOMME_build.csh

```diff
@@ -0,0 +1,35 @@
+#!/bin/csh -f
+
+./Tools/check_lockedfiles || exit -1
+
+setenv COMPILER `./xmlquery COMPILER -value`
+source env_mach_specific || exit -1
+
```

Comment:
This test is a little outside the scope of system testing - it's more like a unit test for the homme dycore.   I don't think that it would work on the cesm side.   Perhaps it would be more appropriate in the unit testing framework.